src/cli/cli.cpp

```diff
@@ -312,9 +314,20 @@ otError Interpreter::ProcessUserCommands(Arg aArgs[])
 
 void Interpreter::SetUserCommands(const otCliCommand *aCommands, uint8_t aLength, void *aContext)
 {
-    mUserCommands        = aCommands;
-    mUserCommandsLength  = aLength;
-    mUserCommandsContext = aContext;
+    otCliCommandList *cmdList = AllocateCommandList();
+    if (cmdList == nullptr)
+    {
+        OutputLine("Not enough entries configured to register user commands");
```

Comment:
Better to log and not send to the CLI output?

Reply:
Should use `otLogCli()` here

Reply:
Or maybe even better to return an error ?

Reply:
Yes. I think so. 
The `otCliSetUserCommands` is defined in `include/opnethread/cli.h` (which is public OT header).